### PR TITLE
Implement RFC 580

### PR DESCRIPTION
**Pull request by aturon**
This commit implements RFC 580 by renaming:
- DList -> LinkedList
- Bitv -> BitVec
- BitvSet -> BitSet
- RingBuf -> VecDeque

More details are in [the RFC](https://github.com/rust-lang/rfcs/pull/580)

[breaking-change]

r? @Gankro 

cc @CloudiDust

### CODE REVIEW

File: src/libcollections/vec_deque.rs [hunk 1]
```diff
@@ -32,12 +32,16 @@ use core::cmp;
...
 
+#[deprecated(since = "1.0.0", reason = "renamed to VecDeque")]
+#[unstable(feature = "collections")]
+pub use VecDeque as RingBuf;
+
 static INITIAL_CAPACITY: usize = 7; // 2^3 - 1
 static MINIMUM_CAPACITY: usize = 1; // 2 - 1
 
-/// `RingBuf` is a circular buffer, which can be used as a double-ended queue efficiently.
+/// `VecDeque` is a circular buffer, which can be used as a double-ended queue efficiently.
```
Comment (tbu-) [inline src/libcollections/vec_deque.rs]: Could you add the word "ring buffer" to the documentation so you can find this collection if you want something akin to a ring buffer?

**Comment by Manishearth** [comment]
Needs a rebase, sorry.

**Comment by aturon** [comment]
Rebased and ready for review.

**Comment by Gankro** [comment]
Merged into #22491

File: src/libstd/collections/mod.rs [hunk 2]
```diff
@@ -109,17 +109,17 @@
 //! |         | get(i)         | insert(i)       | remove(i)      | append | split_off(i)   |
 //! |---------|----------------|-----------------|----------------|--------|----------------|
 //! | Vec     | O(1)           | O(n-i)*         | O(n-i)         | O(m)*  | O(n-i)         |
-//! | RingBuf | O(1)           | O(min(i, n-i))* | O(min(i, n-i)) | O(m)*  | O(min(i, n-i)) |
-//! | DList   | O(min(i, n-i)) | O(min(i, n-i))  | O(min(i, n-i)) | O(1)   | O(min(i, n-i)) |
-//! | Bitv    | O(1)           | O(n-i)*         | O(n-i)         | O(m)*  | O(n-i)         |
+//! | VecDeque | O(1)           | O(min(i, n-i))* | O(min(i, n-i)) | O(m)*  | O(min(i, n-i)) |
+//! | LinkedList   | O(min(i, n-i)) | O(min(i, n-i))  | O(min(i, n-i)) | O(1)   | O(min(i, n-i)) |
+//! | BitVec    | O(1)           | O(n-i)*         | O(n-i)         | O(m)*  | O(n-i)         |
```
Comment (Gankro) [inline src/libstd/collections/mod.rs]: RIP nice computer-generated ASCII table :sob:

**Pull request closed without merging**